src/test/java/org/apache/sysds/test/functions/transform/EncoderSerializationTest.java

```diff
@@ -0,0 +1,143 @@
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class EncoderSerializationTest extends AutomatedTestBase
+{
+	private final static int rows = 2791;
+	private final static int cols = 8;
+
+	private final static Types.ValueType[] schemaStrings = new Types.ValueType[]{
+		Types.ValueType.STRING, Types.ValueType.STRING, Types.ValueType.STRING, Types.ValueType.STRING,
+		Types.ValueType.STRING, Types.ValueType.STRING, Types.ValueType.STRING, Types.ValueType.STRING};
+	private final static Types.ValueType[] schemaMixed = new Types.ValueType[]{
+		Types.ValueType.STRING, Types.ValueType.FP64, Types.ValueType.INT64, Types.ValueType.BOOLEAN,
+		Types.ValueType.STRING, Types.ValueType.FP64, Types.ValueType.INT64, Types.ValueType.BOOLEAN};
+
+
+	public enum TransformType {
+		RECODE,
+		DUMMY
+	}
+
+	@Override
+	public void setUp() {
+		TestUtils.clearAssertionInformation();
+	}
+
+	@Test
+	public void testComposite1() { runTransformSerTest(TransformType.DUMMY, schemaStrings); }
+
+	@Test
+	public void testComposite2() { runTransformSerTest(TransformType.RECODE, schemaMixed); }
+
+	@Test
+	public void testComposite3() { runTransformSerTest(TransformType.RECODE, schemaStrings); }
+
+	@Test
+	public void testComposite4() { runTransformSerTest(TransformType.DUMMY, schemaMixed); }
+
+
+	private void runTransformSerTest(TransformType type, Types.ValueType[] schema) {
+		//data generation
+		double[][] A = getRandomMatrix(rows, cols, -10, 10, 0.9, 8234);
+
+		//init data frame
+		FrameBlock frame = new FrameBlock(schema);
+
+		//init data frame
+		Object[] row = new Object[schema.length];
+		for( int i=0; i < rows; i++) {
+			for( int j=0; j<schema.length; j++ )
+				A[i][j] = UtilFunctions.objectToDouble(schema[j],
+					row[j] = UtilFunctions.doubleToObject(schema[j], A[i][j]));
+			frame.appendRow(row);
+		}
+
+		String spec = "";
+		if(type == TransformType.DUMMY)
+			spec = "{\n \"ids\": true\n, \"dummycode\":[ 2, 7, 8, 1 ]\n\n}";
+		else if(type == TransformType.RECODE)
+			spec = "{\n \"ids\": true\n, \"recode\":[ 2, 7, 1, 8 ]\n\n}";
+
+
+		frame.setSchema(schema);
+		String[] cnames = frame.getColumnNames();
+
+		Encoder encoderIn = EncoderFactory.createEncoder(spec, cnames, frame.getNumColumns(), null);
+		EncoderComposite encoderOut;
+
+		// serialization and deserialization
+		encoderOut = (EncoderComposite) serializeDeserialize(encoderIn);
+		// compare
+		Assert.assertArrayEquals(encoderIn.getColList(), encoderOut.getColList());
+		Assert.assertEquals(encoderIn.getNumCols(), encoderOut.getNumCols());
+
+		List<Encoder> eListIn = ((EncoderComposite) encoderIn).getEncoders();
+		List<Encoder> eListOut = encoderOut.getEncoders();
+		for(int i = 0; i < eListIn.size();  i++) {
+			Assert.assertThat(eListIn.get(i).getColList(), is(eListOut.get(i).getColList()));
+			Assert.assertThat(eListIn.get(i).getNumCols(), is(eListOut.get(i).getNumCols()));
```

Comment:
what does the "is" command do?

otherwise i would merge this commit

Reply:
Hi @Baunsgaard, I used it to compare arrays and long values (like "assertThat(eListIn.get(i).getColList(), equalTo(eListOut.get(i).getColList())))"), but now replaced by simple "assertArrayEquals" and "assertEquals".